crates/ursa-gateway/src/indexer/mod.rs

```diff
@@ -1 +1,56 @@
 pub mod model;
+
+use crate::indexer::model::IndexerResponse;
+use libp2p::multiaddr::Protocol;
+use rand::Rng;
+use std::net::SocketAddrV4;
+use tracing::{error, warn};
+
+// Randomly chooses a provider and returns its addresses.
+pub fn get_provider(indexer_response: IndexerResponse) -> Option<Vec<SocketAddrV4>> {
+    // We expect that MultihashResults will have at most 1 element.
+    let providers = match indexer_response.multihash_results.get(0) {
+        Some(result) => &result.provider_results,
+        None => {
+            error!("indexer result did not contain a multi-hash result");
+            return None;
+        }
+    };
+
+    if !providers.is_empty() {
```

Comment:
should we also filter by metadata "FleekNetwork" ?
https://github.com/fleek-network/ursa/blob/9d8d626aa414c8af03233f61749eb94549df1a4c/crates/ursa-index-provider/src/advertisement.rs#L54

Reply:
@theBeardA Yeah, true, good catch.

@kckeiks Maybe let me handle the filtering in another PR.

Reply:
Will handle in different PR